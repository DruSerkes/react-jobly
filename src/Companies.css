.Companies {
	margin: 2rem;
}

.Companies li {
	list-style-type: none;
}

.Companies-Sorry {
	margin: 5rem auto;
	font-size: 1.6rem;
	font-weight: 300;
}
