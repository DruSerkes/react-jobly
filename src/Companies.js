import React, { useState, useEffect } from 'react';
import JoblyApi from './JoblyApi';
import CompanyCard from './CompanyCard';
// import './Companies.css';

const Companies = () => {
	const [ companies, setCompanies ] = useState(null);

	useEffect(() => {
		const getCompanies = async () => {
			let companies = await JoblyApi.getAllCompanies();
			setCompanies([ ...companies ]);
		};
		getCompanies();
	}, []);

	if (!companies) {
		return <h2>Loading Companies &hellip;</h2>;
	}

	return (
		<div className="Companies">
			{/* {companies.map((company) => <p>{company.name}</p>)} */}
			{companies.map((company) => <CompanyCard company={company} key={company.handle} />)}
		</div>
	);
};

export default Companies;
