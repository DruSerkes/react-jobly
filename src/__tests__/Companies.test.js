import React from 'react';
import { render } from '@testing-library/react';
import Companies from '../Companies';
import { MemoryRouter, Route } from 'react-router-dom';

describe('Companies tests', () => {
	it('renders without crashing', () => {
		render(
			<MemoryRouter initialEntries={[ '/companies/' ]}>
				<Route exact path="/companies/">
					<Companies />
				</Route>
			</MemoryRouter>
		);
	});

	it('matches snapshot', () => {
		const { asFragment } = render(
			<MemoryRouter initialEntries={[ '/companies/' ]}>
				<Route exact path="/companies/">
					<Companies />
				</Route>
			</MemoryRouter>
		);
		expect(asFragment()).toMatchSnapshot();
	});
});
