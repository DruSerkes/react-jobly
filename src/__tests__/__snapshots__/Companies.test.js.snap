// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Companies tests matches snapshot 1`] = `
<DocumentFragment>
  <h2>
    Loading Companies â€¦
  </h2>
</DocumentFragment>
`;
